storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -193,7 +192,16 @@ public File cacheDir() {
     public void remove(Uuid key) {
         lock.readLock().lock();
         try {
-            internalCache.invalidate(key);
+            internalCache.asMap().computeIfPresent(key, (k, v) -> {
+                try {
+                    v.markForCleanup();
+                    expiredIndexes.put(v);
```

Comment:
@showuon  @jeel2420  
Should not we use expiredIndxes.offer instead of expiredIndexes.put()
as put will block the operation if queue size is full ?

Reply:
Use `offer`. Thanks.

Reply:
Do something like we did in the other place:
```
if (!expiredIndexes.offer(entry)) {
                            log.error("Error while inserting entry {} for key {} into the cleaner queue because queue is full.", entry, key);
                        }
```

Reply:
@showuon @kamalcph  @jeel2420  I think we are entering into a deadlock state here 
During remove we try to take a readLock  but in markForCleanUp  we try to take a write lock , Will it not result a deadlock state ?

Reply:
>  I was just thinking around it , if this fails , it will create a dump of files with delete suffix entry which never gets deleted from the disk ? Is the behaviour ok ?

It's OK, it will be deleted in next startup. Of course we can have a way to retry enqueue them, but I think it's a pretty rare case, we can keep it as is. 

> During remove we try to take a readLock but in markForCleanUp we try to take a write lock , Will it not result a deadlock state ?

The readlock in `remove` method is `RemoteIndexCache#lock`. The writeLock in `markForCleanUp` method is `Entry#lock`. I agree it's confusing to have the same name within the same parent class. Could you file another PR to update the lock variable name?

Reply:
Yah sure @showuon  Let me do it. Should I create a separate ticket or another PR attached with the  same ticket ?

Reply:
Another PR with `MINOR` prefixed.